homeassistant/components/shopping_list/__init__.py

```diff
@@ -53,10 +55,24 @@
 )
 
 
-@asyncio.coroutine
-def async_setup(hass, config):
+async def async_setup(hass, config):
     """Initialize the shopping list."""
 
+    # Only import if we haven't before.
+    if not hass.config_entries.async_entries(DOMAIN):
```

Comment:
I don't think you should add this, it should not setup if already.

Do you have a bug without ?

Reply:
It's indeed working without it, was just copying it over like that from another component

Reply:
Was too fast, it is NOT working without that check. Otherwise, two entries would be created.

Reply:
Since we're setting a unique_id in the config flow, we should let the check in the config flow abort. Please remove the check here.

Reply:
When I wrote that comment, the unique_id thing was not implemented. Now its working fine without those lines